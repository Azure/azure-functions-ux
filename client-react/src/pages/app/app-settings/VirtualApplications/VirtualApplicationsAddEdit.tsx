import * as React from 'react';
import { TextField } from 'office-ui-fabric-react/lib-commonjs/TextField';
import { VirtualApplication } from '../../../../models/WebAppModels';
import { Toggle } from 'office-ui-fabric-react/lib-commonjs/Toggle';

export interface HandlerMappingAddEditProps extends VirtualApplication {
  updateVirtualApplication: (item: VirtualApplication) => any;
}

const VirtualApplicationsAddEdit: React.SFC<HandlerMappingAddEditProps> = props => {
  const { updateVirtualApplication, children, ...virtualApplication } = props;
  const updatePhysicalPath = (physicalPath: string) => {
    updateVirtualApplication({
      ...virtualApplication,
      physicalPath,
    });
  };

  const updateVirtualPath = (virtualPath: string) => {
    updateVirtualApplication({ ...virtualApplication, virtualPath });
  };

  const updateVirtualDirectory = (virtualDirectory: boolean) => {
    updateVirtualApplication({
      ...virtualApplication,
      virtualDirectory,
    });
  };
  const updatePreloadEnabled = (preloadEnabled: boolean) => {
    updateVirtualApplication({
      ...virtualApplication,
      preloadEnabled,
    });
  };
  return (
    <div>
      <TextField label="Physical Path" id="extension" value={virtualApplication.physicalPath} onChanged={updatePhysicalPath} />
      <TextField label="Virtual Path" id="value" value={virtualApplication.virtualPath} onChanged={updateVirtualPath} />
      <Toggle
        label="Directory or Application"
        id="sticky"
        defaultChecked={virtualApplication.virtualDirectory}
        onChanged={updateVirtualDirectory}
        onText="Directory"
        offText="Application"
      />
      {virtualApplication.virtualDirectory ? null : (
        <Toggle
          label="Preload Enabled"
          id="preloadEnabled"
          defaultChecked={virtualApplication.preloadEnabled}
          onChanged={updatePreloadEnabled}
          onText="On"
          offText="Off"
        />
      )}
    </div>
  );
};

export default VirtualApplicationsAddEdit;
