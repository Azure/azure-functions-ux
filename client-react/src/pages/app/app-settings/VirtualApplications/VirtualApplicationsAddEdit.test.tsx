import { shallow } from 'enzyme';
import * as React from 'react';
import VirtualApplicationsAddEdit from './VirtualApplicationsAddEdit';
const VirtualApplicationsAddEditAny = VirtualApplicationsAddEdit as any;
describe('Virtual Applications Add/Edit', () => {
  describe('Shallow Testing', () => {
    let container;

    beforeEach(() => {
      container = shallow(<VirtualApplicationsAddEditAny values={{ virtualApplications: [] }} />);
    });

    it('renders without crashing', () => {
      expect(container.length).toBe(1);
    });
  });
});
