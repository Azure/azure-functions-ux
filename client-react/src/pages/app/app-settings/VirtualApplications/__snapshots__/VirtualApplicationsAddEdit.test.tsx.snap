// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Virtual Applications Add/Edit Shallow Testing render matches snapshot: DefaultSnapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VirtualApplicationsAddEdit
    values={
      Object {
        "virtualApplications": Array [],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <StyledTextFieldBase
          id="extension"
          label="Physical Path"
          onChanged={[Function]}
          value={undefined}
        />,
        <StyledTextFieldBase
          id="value"
          label="Virtual Path"
          onChanged={[Function]}
          value={undefined}
        />,
        <StyledToggleBase
          defaultChecked={undefined}
          id="sticky"
          label="Directory or Application"
          offText="Application"
          onChanged={[Function]}
          onText="Directory"
        />,
        <StyledToggleBase
          defaultChecked={undefined}
          id="preloadEnabled"
          label="Preload Enabled"
          offText="Off"
          onChanged={[Function]}
          onText="On"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "id": "extension",
          "label": "Physical Path",
          "onChanged": [Function],
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "id": "value",
          "label": "Virtual Path",
          "onChanged": [Function],
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "defaultChecked": undefined,
          "id": "sticky",
          "label": "Directory or Application",
          "offText": "Application",
          "onChanged": [Function],
          "onText": "Directory",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "defaultChecked": undefined,
          "id": "preloadEnabled",
          "label": "Preload Enabled",
          "offText": "Off",
          "onChanged": [Function],
          "onText": "On",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <StyledTextFieldBase
            id="extension"
            label="Physical Path"
            onChanged={[Function]}
            value={undefined}
          />,
          <StyledTextFieldBase
            id="value"
            label="Virtual Path"
            onChanged={[Function]}
            value={undefined}
          />,
          <StyledToggleBase
            defaultChecked={undefined}
            id="sticky"
            label="Directory or Application"
            offText="Application"
            onChanged={[Function]}
            onText="Directory"
          />,
          <StyledToggleBase
            defaultChecked={undefined}
            id="preloadEnabled"
            label="Preload Enabled"
            offText="Off"
            onChanged={[Function]}
            onText="On"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "id": "extension",
            "label": "Physical Path",
            "onChanged": [Function],
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "id": "value",
            "label": "Virtual Path",
            "onChanged": [Function],
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "defaultChecked": undefined,
            "id": "sticky",
            "label": "Directory or Application",
            "offText": "Application",
            "onChanged": [Function],
            "onText": "Directory",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "defaultChecked": undefined,
            "id": "preloadEnabled",
            "label": "Preload Enabled",
            "offText": "Off",
            "onChanged": [Function],
            "onText": "On",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
